bin/propolis-server/src/main.rs

```diff
@@ -86,6 +86,27 @@ async fn main() -> anyhow::Result<()> {
                 |error| anyhow!("failed to create logger: {}", error),
             )?;
 
+            // Check that devices conform to expected API version
+            use propolis::api_version;
+            if let Err(e) = api_version::check() {
+                match e {
+                    api_version::Error::Io(io_err) => {
+                        slog::error!(log, "Error checking API versions");
+                        return Err(io_err.into());
+                    }
+                    api_version::Error::Mismatch(comp, act, exp) => {
+                        slog::error!(
+                            log,
+                            "Inadequate {} API version {} < {}",
+                            comp,
+                            act,
+                            exp
+                        );
+                        return Err(e.into());
+                    }
+                }
```

Comment:
You already have a Display impl so can just use that for the log too:
```suggestion
                slog::error!(log, "API version check failed: {}", e);
                return Err(e.into());
```

Reply:
Or alternatively just skip the log altogether and do something like mentioned in the standalone case.

Reply:
Yeah, I just went to the `context()` case